### ISSUE TITLE
Help with identifying good drives

### ISSUE THREAD

**Issue by SeanRamey** [superg/redumper#144]
Hi. So, I've been trying to make some back up discs of my PC games, as much a 1:1 copy as possible. I've run into some issues with 1503 AD which seems to use SafeDisc 2.80, and I've been using cdrdao on Linux to do my backing up. I opened an issue with cdrdao and me and another guy have been learning that the problem is certain drives can't seem to write all data properly. I run across this project in my research (redump) and I was thinking maybe you could help me identify drives that can write these copy protections properly. [Here is the link ](https://github.com/cdrdao/cdrdao/issues/28)to the cdrdao issue.

Thanks!

**Comment by superg** [comment]
http://redump.org/disc/64375/

SafeDisc protection intentionally corrupts some files so standard drive reads slowdown/error on a disc copy.
To dump such games you need a drive that reads data tracks as audio (scrambled mode) and you need software that recognizes the protection and treats it appropriately (redumper). 

Compatible drives: http://wiki.redump.org/index.php?title=Optical_Disc_Drive_Compatibility

TL;DR: get good plextor or asus and use redumper.

**Comment by SeanRamey** [comment]
> http://redump.org/disc/64375/
> 
> SafeDisc protection intentionally corrupts some files so standard drive reads slowdown/error on a disc copy. To dump such games you need a drive that reads data tracks as audio (scrambled mode) and you need software that recognizes the protection and treats it appropriately (redumper).
> 
> Compatible drives: http://wiki.redump.org/index.php?title=Optical_Disc_Drive_Compatibility
> 
> TL;DR: get good plextor or asus and use redumper.

Yes, I know how to get a good dump/read, but my question was about which drives are good for **_writing_** .

The same drives that are fine for reading are not necessarily good for writing, and the use for disc image files is limited if they can't be properly burned back to a disc.

**Comment by superg** [comment]
There are a lot of issues with writing. One thing is that you will not be able to write back SafeDisc errors because there is no way to simulate C2 errors with consumer optical drive.
Second, when you burn image, your write offset will be different, TOC and subchannel will be regenerated. 
Writing is complex and is definitely out of redumper scope.

**Comment by SeanRamey** [comment]
> There are a lot of issues with writing. One thing is that you will not be able to write back SafeDisc errors because there is no way to simulate C2 errors with consumer optical drive.

What is a C2 error, what is a SafeDisc error, and what is the difference between them? Thus far I have been able to seemingly make 1:1 working copies of some earlier SafeDisc games, so it seems to me that what you say is not true unless I misunderstand.

>Second, when you burn image, your write offset will be different, TOC and subchannel will be regenerated. Writing is complex and is definitely out of redumper scope.

I do realize that writing seems to be out of redumper's scope. I'm just looking for information and help from people who seem to have the most knowledge about these topics.

**Comment by superg** [comment]
> What is a C2 error, what is a SafeDisc error, and what is the difference between them? Thus far I have been able to seemingly make 1:1 working copies of some earlier SafeDisc games, so it seems to me that what you say is not true unless I misunderstand.
C2 error is a notification from a drive that data is corrupted (there are checksums on a lowest level to know that).

1. C2 errors from the scratched disc / disc degradation are what I refer to as genuine errors.
2. SafeDisc protection intentionally masters disc with C2 errors so you cannot easily dump the disc, let's call them fake errors.

The distinction between (1) and (2) is important because (1) is often repairable (constant re-reads might eventually get the correct data), but (2) was mastered that way and drive will not be able to fix it.
In the worst case, when you dump scratched SafeDisc, you have a mix of both (1) and (2) and you need to be able to tell what is what, because it's pointless to re-read (2).

> Thus far I have been able to seemingly make 1:1 working copies of some earlier SafeDisc games, so it seems to me that what you say is not true unless I misunderstand.

You can make a copy of SafeDisc game but it will not be 1:1 because of mentioned (2). 1:1 doesn't mean playability, 1:1 means the exact disc copy (dump), minus of course the mastering errors.
I suggest to go to redump.org wiki and read the dumping guides. There is plenty of information that will be relevant.
